test/suite/test_compact12.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wttest
+from wiredtiger import stat
+
+kilobyte = 1024
+
+# test_compact12.py
+# This test creates:
+#
+# - One table with the first 1/4 of keys deleted and obsolete and the last 10th of the keys
+#   removed with fast truncate but not yet obsolete.
+#
+# It checks that:
+#
+# - Compaction correctly rewrites pages in WT_REF_DELETED state but are still on disk.
+class test_compact12(wttest.WiredTigerTestCase):
+    create_params = 'key_format=i,value_format=S,allocation_size=4KB,leaf_page_max=32KB,leaf_value_max=16MB'
+    conn_config = 'cache_size=10MB,statistics=(all),verbose=[compact:4]'
```

Comment:
Is there any reason why we have such a low cache size value? The default of 100MB should not have any impact or does it?

Reply:
I was playing around with this cache in testing, there's no need to leave it at 10MB, I'll change it back to the default.

Reply:
Fixed https://github.com/wiredtiger/wiredtiger/commit/65ddbcaa5af07a9f0111a571f266111140c411cb